clippy_lints/src/methods/collapsible_str_replace.rs

```diff
@@ -0,0 +1,275 @@
+// run-rustfix
+
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::get_parent_expr;
+use clippy_utils::visitors::for_each_expr;
+use core::ops::ControlFlow;
+use if_chain::if_chain;
+use rustc_ast::ast::LitKind;
+use rustc_data_structures::fx::FxHashSet;
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_hir::{ExprKind, Path, QPath};
+use rustc_lint::LateContext;
+use rustc_middle::ty;
+use rustc_span::source_map::Spanned;
+use rustc_span::Span;
+
+use super::method_call;
+use super::COLLAPSIBLE_STR_REPLACE;
+
+pub(super) fn check<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx hir::Expr<'tcx>,
+    name: &str,
+    recv: &'tcx hir::Expr<'tcx>,
+) {
+    if name == "replace" {
+        // The receiver of the method call must be `str` type to lint `collapsible_str_replace`
+        let original_recv = find_original_recv(recv);
+        let original_recv_ty_kind = cx.typeck_results().expr_ty(original_recv).peel_refs().kind();
+        let original_recv_is_str_kind = matches!(original_recv_ty_kind, ty::Str);
+
+        if_chain! {
+            if original_recv_is_str_kind;
+            if let Some(parent) = get_parent_expr(cx, expr);
+            if let Some((name, ..)) = method_call(parent);
+            if name == "replace";
```

Comment:
I noticed that this doesn't really work. You'll also have to compare if the `to` argument is the same as the one of the last `replace` call in the current chain. Otherwise this wouldn't get linted:

```
.replace(_, "l").replace(_, "l").replace(_, "x")
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```